### PR TITLE
gh-101143: Remove references to TimerHandle from asyncio.base_events.BaseEventLoop._add_callback

### LINKED ISSUE #101143: asyncio.base_events.BaseEventLoop._add_callback references TimerHandle but never uses them
# Bug report

After 592ada9b4b08ad57037e365b9c462d71c96e4453 it appears this function not longer cares about `self._scheduled` or `TimerHandle` but the docstring seems to imply otherwise

# Your environment

- CPython versions tested on: 3.12.0a4
- Operating system and architecture: MacOS

I only noticed because it was unexpectedly more expensive

unexpectedly_more_expensive

I think its dead code and can be simplified to:

```diff
diff --git a/Lib/asyncio/base_events.py b/Lib/asyncio/base_events.py
index cbabb43ae0..070ea1044d 100644
--- a/Lib/asyncio/base_events.py
+++ b/Lib/asyncio/base_events.py
@@ -1857,11 +1857,9 @@ def call_exception_handler(self, context):
                                  exc_info=True)
 
     def _add_callback(self, handle):
-        """Add a Handle to _scheduled (TimerHandle) or _ready."""
+        """Add a Handle to _ready."""
         assert isinstance(handle, events.Handle), 'A Handle is required here'
-        if handle._cancelled:
-            return
-        assert not isinstance(handle, events.TimerHandle)
+        if not handle._cancelled:
             self._ready.append(handle)
 
     def _add_callback_signalsafe(self, handle):
```

### Linked PRs
* gh-101197
* gh-101216
* gh-101217

### PR DESCRIPTION
**Pull request by bdraco**
gh-101143: Remove references to TimerHandle from asyncio.base_events.BaseEventLoop._add_callback

After 592ada9b4b08ad57037e365b9c462d71c96e4453 it appears this function no longer cares about `self._scheduled` or `TimerHandle` but the docstring seems to imply otherwise

fixes #101143

### CODE REVIEW

**Comment by bdraco** [comment]
Tried to sign the CLA but getting `Internal Server Error`

File: Lib/asyncio/base_events.py [hunk 1]
```diff
@@ -1857,12 +1857,10 @@ def call_exception_handler(self, context):
                                  exc_info=True)
 
     def _add_callback(self, handle):
-        """Add a Handle to _scheduled (TimerHandle) or _ready."""
+        """Add a Handle to _ready."""
         assert isinstance(handle, events.Handle), 'A Handle is required here'
```
Comment (gvanrossum) [inline Lib/asyncio/base_events.py]: I wonder if we should drop this assertion as well? It's still going to be the slowest part of the method.

**Review by gvanrossum — COMMENTED** [review]
I'm guessing that since you're deleting only dead code this doesn't need a new test.

File: Lib/asyncio/base_events.py (hunk 1 above)
Comment (bdraco) [reply Lib/asyncio/base_events.py]: Its only in internals and already covered by tests so the risk of implementation error seems low. Will push another commit to remove it.

Comment (bdraco) [reply Lib/asyncio/base_events.py]: Removed in 7ee5f56e9b8ce8329ff3387314848affb783b83f

**Review by gvanrossum — APPROVED** [review]
Works for me. I'd like @kumaraditya303 to take a look first though. Kumar, if you approve just merge it. Also, should we backport this to 3.11? Or even 3.10? The change that made it all pointless is pretty old IIUC.

**Comment by bdraco** [comment]
> Also, should we backport this to 3.11? Or even 3.10? The change that made it all pointless is pretty old IIUC.

Backport to 3.10 (where I saw this in production)/3.11 would be great since this has a real world impact for Home Assistant users processing d-bus messages especially on the slower RPis.

**Review by kumaraditya303 — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2023-01-20-10-46-59.gh-issue-101143.hJo8hu.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Remove unused references to TimerHandle in
+asyncio.base_events.BaseEventLoop._add_callback
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2023-01-20-10-46-59.gh-issue-101143.hJo8hu.rst]:
```suggestion
Remove unused references to :class:`~asyncio.TimerHandle` in
``asyncio.base_events.BaseEventLoop._add_callback``.
```

**Comment by kumaraditya303** [comment]
> Also, should we backport this to 3.11? Or even 3.10? The change that made it all pointless is pretty old IIUC.

Let's backport it.

**Comment by bdraco** [comment]
Thanks for taking the time to review this.

**Comment by gvanrossum** [comment]
Thanks for the PR! It’s always great to _delete_ code, we need more of that.

**Pull request merged**